model/project_ref.go

```diff
@@ -404,6 +404,47 @@ func (p *ProjectRef) GetPatchTriggerAlias(aliasName string) (patch.PatchTriggerD
 	return patch.PatchTriggerDefinition{}, false
 }
 
+func (p *ProjectRef) GetProjectParserMergedProjectRef(version string) *ProjectRef {
+	lookupVersion := false
+	if version == "" {
+		lastGoodVersion, err := FindVersionByLastKnownGoodConfig(p.Id, -1)
+		if err != nil || lastGoodVersion == nil {
+			grip.Error(message.WrapError(err, message.Fields{
+				"message":    fmt.Sprintf("Unable to retrieve last good version for project '%s'", p.Id),
+				"project_id": p.Id,
+				"version":    version,
+			}))
+			return p
+		}
+		version = lastGoodVersion.Id
+		lookupVersion = true
+	}
+	parserProject, err := ParserProjectFindOneById(version)
+	if err != nil {
+		grip.Debug(message.Fields{
+			"message":        fmt.Sprintf("Error retrieving parser project for version '%s'", version),
+			"project_id":     p.Id,
+			"version":        version,
+			"lookup_version": lookupVersion,
+			"err":            err.Error(),
+		})
+		return p
+	}
+	pRef := p
```

Comment:
Can we write some sort of test to verify that this doesn't destructively modify the input? (Since this is pointers I think it will)

Reply:
This does destructively modify the input.  However isn't that what we are aiming to do here?

Reply:
Gotcha -- we can do that; typically when a method returns the data structure, it's implied that the input isn't changed, and we return a modified version. If we _intend_ to modify p, then there's no reason to return a different projectRef; this function can just return an error.